sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumerOptions.cs

```diff
@@ -111,6 +111,22 @@ public int PrefetchCount
             }
         }
 
+        /// <summary>
+        ///   Indicates whether or not the consumer requests metrics for the state of its
+        ///   partition each time that events are received.
+        /// </summary>
+        ///
+        /// <value><c>true</c> if metrics are requested when events are received; otherwise, <c>false</c>.</value>
+        ///
+        /// <remarks>
+        ///   When metrics are requested, each event received from the Event Hubs service will carry metadata
+        ///   about the state of a partition that it otherwise would not.  This results in a small amount of
+        ///   additional network bandwidth consumption that is generally a favorable trade-off when considered
+        ///   against periodically making requests for partition properties using the Event Hub client.
+        /// </remarks>
+        ///
+        public bool RequestPartitionMetricUpdates { get; set; } = true;
```

Comment:
The last we discussed with the service team, we wanted keep the name of this option very specific to the exact task it was related to i.e fetching information around the last enqueued event instead of the generic "metrics" term.

I understand that keeping a general name does help in keeping this option open in the future to get other "metrics" in the future, but am not sure if we want to encourage that.

Keeping a specific name makes it very clear to the user as to what the option would do and avoid using the piggy back mechanism for other features.

We (@JeffreyRichter and I) wanted to propose `TrackLastEnqueuedEventInfo`.

Reply:
We're putting this in unplanned at the 11th hour.   I'm happy to have the naming discussion and make adjustments for preview 4.   I am intentionally making unilateral decisions to avoid blocking code complete.

Reply:
And as an extension to the above, we (@JeffreyRichter and I) want to propose `LastEnqueuedEventInfo` instead of `PartitionMetrics` as the name of the property on the consumer.

This way its more in your face that the option (`TrackLastEnqueuedEventInfo`) and the property (`LastEnqueuedEventInfo`) are tightly tied together

Reply:
Going to park the naming considerations and revisit in preview 4.   I'll open an issue later today for design proposals and adjustments.

Reply:
I agree that this is unplanned and at the last minute. I would still request to please
- rename the user facing boolean option to `TrackLastEnqueuedEventInfo`
- rename the user facing property on consumer to `LastEnqueuedEventInfo`

We will follow the same for Javascript.

The class name `PartitionConsumerMetrics` can stay. We can use the preview 4 to have the naming discussion.

Reply:
Surface area for this looks to be manageable without causing too much churn.   I'll adopt the suggestions, but am expanding `Info` to `Information` for internal consistency with similar types in the library.

Reply:
Renamed as follows:

- `PartitionConsumerMetrics` => `LastEnqueuedEventProperties`    _(properties used over information for consistency with other metadata types in the library)_
- `EventHubConsumerOptions.RequestPartitionMetricUpdates` => `EventHubConsumerOptions.TrackLastEnqueuedEventInformation`
- `EventHubConsumer.PartitionMetrics` => `EventHubConsumer.LastEnqueuedEventInformation`